pybossa/api/task_run.py

```diff
@@ -75,12 +74,13 @@ def _ensure_task_was_requested(self, task, guard):
             raise Forbidden('You must request a task first!')
 
     def _add_user_info(self, taskrun):
-        if current_user.is_anonymous():
-            taskrun.user_ip = request.remote_addr
-            if taskrun.user_ip is None:
-                taskrun.user_ip = '127.0.0.1'
-        else:
-            taskrun.user_id = current_user.id
+        if not taskrun.user_ip and not taskrun.user_id:
```

Comment:
hold on with this PR, this line is security flaw

Reply:
OK, we put some additional thoughts into this line. It enables any service using method `create_task_run()` to override `user_ip` and `user_id` which is fine. This enables pbclient to send user IP or external UID via API and enables mocking user data. IMO it's fine

Reply:
The issue is that a user can fake user IDs, correct? This will be a "problem" from the point of view of allowing anyone to create as many fake taskruns as they want. I would love to discuss about this. Can you join our Gitter room?